crates/ruff_python_formatter/src/context.rs

```diff
@@ -1,11 +1,22 @@
 use crate::comments::Comments;
-use crate::string::QuoteChar;
+use crate::string::{QuoteChar, StringQuotes};
 use crate::PyFormatOptions;
 use ruff_formatter::{Buffer, FormatContext, GroupId, IndentWidth, SourceCode};
 use ruff_source_file::Locator;
 use std::fmt::{Debug, Formatter};
 use std::ops::{Deref, DerefMut};
 
+/// Location of the expression which is currently being formatted.
+#[derive(Copy, Clone, Debug, Default)]
+pub(crate) enum ExpressionLocation {
+    /// The expression is inside an f-string, in the replacement field i.e., `f"foo {x}"`.
+    ///
+    /// The containing `StringQuotes` is surrounding f-string quote information.
+    InsideFString(StringQuotes),
+    #[default]
+    Other,
+}
+
 #[derive(Clone)]
 pub struct PyFormatContext<'a> {
```

Comment:
What's the size of `PyFormatContext`. I wonder if the added field pushes it over the limit so that llvm/rust opts out of some optimisations. Or try re-running the benchmarks to see if the performance still regresses.

Reply:
The size is the same before and after (80).